htdocs/bom/class/bom.class.php

```diff
@@ -1470,6 +1465,17 @@ public function calculateCosts()
 					$unitforline = measuringUnitString($line->fk_unit, '', '', 1);
 					$qtyhourforline = convertDurationtoHour($line->qty, $unitforline);
 
+					$line->unit_cost = price2num((!empty($tmpproduct->cost_price)) ? $tmpproduct->cost_price : $tmpproduct->pmp);
+					if (empty($line->unit_cost)) {
+						if ($productFournisseur->find_min_price_product_fournisseur($line->fk_product) > 0) {
+							if ($productFournisseur->fourn_remise_percent != "0") {
+								$line->unit_cost = $productFournisseur->fourn_unitprice_with_discount;
```

Comment:
I did not check, but if fourn_unitprice_with_discount is up-to-date even when there is no discount (i.e., fourn_unit_price == fourn_unitprice_with_discount when there is no discount), then unit_price can always use fourn_unitprice_with_discount without any specific test.

Reply:
Ok, for I actually just copy and pasted from the product cost part of code from the block above. The problem I was having with this was that the service cost was 0 because I use buying price (thirdparty supplies the service). I will dig a bit deeper and check